### PR TITLE
adding fattest.databases module

### PR DESCRIPTION
**Pull request by KyleAure**
Reattempting to get database rotation code-base into a common location.  Originally was going to put into `fattest.simplicity`, but some components require java 8 and we are unable to upgrade the fattest.simplicity component to java 8 at this time.

Instead, I am moving this code-base to it's own project `fattest.databases`.

Original Idea: https://github.com/OpenLiberty/open-liberty/pull/9921

### CODE REVIEW

File: dev/fattest.databases/bnd.bnd [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+Include-Resource: \
+  META-INF=resources/META-INF
+
+test.project: true
+
+-buildpath: \
+	com.ibm.ws.componenttest:public.api;version=1.0.0,\
+	com.ibm.ws.org.osgi.annotation.versioning;version=latest,\
+	org.testcontainers:database-commons;version=1.12.3,\
+	org.testcontainers:jdbc;version=1.12.3,\
```
Comment (KyleAure) [inline dev/fattest.databases/bnd.bnd]: I removed the compile time dependencies on the database specific components by using reflection to create the containers.

File: dev/fattest.databases/bnd.bnd [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	org.testcontainers:testcontainers;version=1.12.3,\
+	fattest.simplicity;version=latest,\
+	org.rnorth.duct-tape:duct-tape;version=1.0.7,\
+	org.slf4j:slf4j-api;version=1.7.7
```
Comment (KyleAure) [inline dev/fattest.databases/bnd.bnd]: These two dependencies are not needed at compile time.  But, eclipse and bnd tools will complain loudly about it with: 
`The type org.slf4j.Logger cannot be resolved. It is indirectly referenced from required .class files`
`The type org.rnorth.ducttape.ratelimits.RateLimiter cannot be resolved. It is indirectly referenced from required .class files`

Comment (KyleAure) [inline dev/fattest.databases/bnd.bnd]: These two dependencies are not needed at compile time.  But, eclipse and bnd tools will complain loudly about it with: 
`The type org.slf4j.Logger cannot be resolved. It is indirectly referenced from required .class files`
`The type org.rnorth.ducttape.ratelimits.RateLimiter cannot be resolved. It is indirectly referenced from required .class files`

File: dev/fattest.databases/resources/META-INF/services/org.testcontainers.dockerclient.DockerClientProviderStrategy [hunk 3]
```diff
@@ -0,0 +1 @@
+componenttest.topology.utils.ExternalTestServiceDockerClientStrategy
```
Comment (aguibert) [inline dev/fattest.databases/resources/META-INF/services/org.testcontainers.dockerclient.DockerClientProviderStrategy]: Is this file needed? I believe the copy in fattest.simplicity should be sufficient

Comment (KyleAure) [reply dev/fattest.databases/resources/META-INF/services/org.testcontainers.dockerclient.DockerClientProviderStrategy]: Removed.

**Comment by KyleAure** [comment]
#build 
#libbybot

**Comment by KyleAure** [comment]
Rebuilding here: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewDefinition&id=_JWdEUD7dEemKFI0xRbWJ_w
#libbybot
